podcasts/DownloadManager+Logging.swift

```diff
@@ -0,0 +1,127 @@
+import PocketCastsDataModel
+
+extension DownloadManager {
+    func logDownload(_ episode: BaseEpisode, failure: FailureReason, extraProperties: [String: Any?] = [:]) {
```

Comment:
Just a question: do we use semantic expression to represent empty values? Like  `extraProperties: [String: Any?] .empty`

Reply:
We don't currently do this, afaik. Would this be a series of extensions on `ExpressibleByXLiteral` with the initializer defined?

```swift
public extension ExpressibleByDictionaryLiteral {
    static var empty: Self {
        return Self()
    }
}
```

Is there any worry in colliding with `_IntentValue` conforming types (like [Array](https://developer.apple.com/documentation/swift/array/empty#)) from AppIntent's [`EmptyResolverSpecification`](https://developer.apple.com/documentation/appintents/emptyresolverspecification#)?

Reply:
>Is there any worry in colliding with _IntentValue conforming types (like [Array](https://developer.apple.com/documentation/swift/array/empty#)) from AppIntent's [EmptyResolverSpecification](https://developer.apple.com/documentation/appintents/emptyresolverspecification#)?

I'm not entirely sure as I've always created the extension from the collections or the type I wanted to use. So far Collections and Strings